tracer/test/Datadog.Trace.Tests/Tagging/ActivityTagsTests.cs

```diff
@@ -0,0 +1,128 @@
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using Datadog.Trace.Activity;
+using Datadog.Trace.Activity.DuckTypes;
+using FluentAssertions;
+using Moq;
+using Xunit;
+
+namespace Datadog.Trace.Tests.Tagging;
+
+[Collection(nameof(ActivityTagsTests))]
+public class ActivityTagsTests
+{
+    public enum TagKind
+    {
+        /// <summary>
+        /// Tag should be found in meta
+        /// </summary>
+        Meta,
+
+        /// <summary>
+        /// Tag should be found in metrics
+        /// </summary>
+        Metric,
+    }
+
+    public static IEnumerable<object[]> TagData =>
+        new List<object[]>()
+        {
+            // key, value, location
+            new object[] { "char_val", 'c', TagKind.Meta },
+            new object[] { "string_val", "val", TagKind.Meta },
+            new object[] { "bool_val", true, TagKind.Meta },
+            new object[] { "byte_val", (byte)5, TagKind.Metric },
+            new object[] { "sbyte_val", (sbyte)-5, TagKind.Metric },
+            new object[] { "short_val", (short)-5, TagKind.Metric },
+            new object[] { "ushort_val", (ushort)5, TagKind.Metric },
+            new object[] { "int_val", -5, TagKind.Metric },
+            new object[] { "uint_val", 5U, TagKind.Metric },
+            new object[] { "long_val", -5L, TagKind.Metric },
+            new object[] { "ulong_val", 5UL, TagKind.Metric },
+            new object[] { "float_val", -5.0f, TagKind.Metric },
+            new object[] { "double_val", -5.0, TagKind.Metric },
+            new object[] { "object_val", new(), TagKind.Meta }
+        };
+
+    public static IEnumerable<object[]> ArrayTagData =>
+        new List<object[]>()
+        {
+            // key, value, location, expected_dot_notation_dict
+            new object[] { "char[]_val", new[] { 'c', 'd' }, TagKind.Meta, new Dictionary<string, object> { { "char[]_val.0", "c" }, { "char[]_val.1", "d" } } },
+            new object[] { "string[]_val", new[] { "val1", "val2" }, TagKind.Meta, new Dictionary<string, object> { { "string[]_val.0", "val1" }, { "string[]_val.1", "val2" } } },
+            new object[] { "bool[]_val", new[] { true, false }, TagKind.Meta, new Dictionary<string, object> { { "bool[]_val.0", "true" }, { "bool[]_val.1", "false" } } },
+            new object[] { "double[]_val", new[] { -5.0, 5.0 }, TagKind.Metric, new Dictionary<string, object> { { "double[]_val.0", -5.0 }, { "double[]_val.1", 5.0 } } },
+            new object[]
+            {
+                "char[][]_val", new[]
+                {
+                    new object[] { 'a', 'b', 'c' },
+                    new object[] { 'd', 'e', 'f' }
+                }, TagKind.Meta, new Dictionary<string, object> { { "char[][]_val.0", """["a","b","c"]""" }, { "char[][]_val.1", """["d","e","f"]""" } }
```

Comment:
I'm surprised the expected value has double quotes in the string like `"a"` and `"b"`. Is that how Newtonsoft serializes them?